dace/libraries/standard/nodes/reduce.py

```diff
@@ -804,6 +803,243 @@ def expansion(node: 'Reduce', state: SDFGState, sdfg: SDFG):
         #return reduce_node.expand(sdfg, state)
 
 
+@dace.library.expansion
+class ExpandReduceFPGA(pm.ExpandTransformation):

```

Comment:
What do you think about calling this something along the lines of `ExpandReduceFPGAPartialReduction`, so we can also have a more efficient non-partial version (in the future, if we want to)?